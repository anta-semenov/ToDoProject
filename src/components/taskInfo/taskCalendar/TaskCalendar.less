@import '../../../styles/variables.less';

.calendar {
  position: relative;
  box-sizing: border-box;
  margin: 10px  0 0 6px;
  width: 100%;
}
.calendar__day-names {
  position: relative;
  box-sizing: border-box;
  width: 100%;

  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: center;
}
.calendar__day-name {
  position: relative;
  box-sizing: border-box;
  width: 14.2857%;
  padding: 5px;

  text-align: center;
  font-size: @font-size-small;
  font-weight: @font-weight-light;
  cursor: pointer;
  color: @color-font-medium;
}

.calendar__week {
  position: relative;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: center;
}
.calendar__day {
  position: relative;
  box-sizing: border-box;
  width: 14.2857%;
  padding: 5px;

  text-align: center;
  font-size: @font-size-basic;
  font-weight: @font-weight-light;
  cursor: pointer;

  &:hover {

  }

  &.is-prev-month,
  &.is-next-month {
    color: @color-font-light;
  }
}
