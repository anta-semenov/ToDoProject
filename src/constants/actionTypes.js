/*
 * Project
 */
export const ADD_PROJECT = 'ADD_PROJECT'
export const REMOVE_PROJECT = 'REMOVE_PROJECT'
export const COMPLETE_PROJECT = 'COMPLETE_PROJECT'
export const EDIT_PROJECT = 'EDIT_PROJECT'
export const REPLACE_PROJECT = 'REPLACE_PROJECT'

/*
 * Task
 */
export const ADD_TASK = 'ADD_TASK'
export const REMOVE_TASK = 'REMOVE_TASK'
export const COMPLETE_TASK = 'COMPLETE_TASK'
export const EDIT_TASK = 'EDIT_TASK'
export const REPLACE_TASK = 'REPLACE_TASK'
export const ADD_TASK_TO_PROJECT = 'ADD_TASK_TO_PROJECT'
export const ADD_TASK_CONTEXT = 'ADD_TASK_CONTEXT'
export const REMOVE_TASK_CONTEXT = 'REMOVE_TASK_CONTEXT'
export const SET_TASK_TODAY = 'SET_TASK_TODAY'
export const START_TASK_TRACKING = 'START_TASK_TRACKING'
export const STOP_TASK_TRACKING = 'STOP_TASK_TRACKING'

/*
 * Context
 */
export const ADD_CONTEXT = 'ADD_CONTEXT'
export const REMOVE_CONTEXT = 'REMOVE_CONTEXT'
export const EDIT_CONTEXT = 'EDIT_CONTEXT'
export const SWITCH_TASK_CONTEXT = 'SWITCH_TASK_CONTEXT'
export const REPLACE_CONTEXT = 'REPLACE_CONTEXT'

/*
* UI State
*/
export const SET_SELECTED_SECTION = 'SET_SELECTED_SECTION'
export const SET_SIDEBAR_SIZE = 'SET_SIDEBAR_SIZE'
export const SET_ACTIVE_ITEM = 'SET_ACTIVE_ITEM'
export const SET_EDITING_SECTION = 'SET_EDITING_SECTION'
export const TOGGLE_TASK_COMPLETED_LATENCY = 'TOGGLE_TASK_COMPLETED_LATENCY'
export const CLEAR_COMPLETED_LATENT_TASKS = 'CLEAR_COMPLETED_LATENT_TASKS'
export const TOGGLE_TASK_LATENCY = 'TOGGLE_TASK_LATENCY'
export const CLEAR_LATENT_TASKS = 'CLEAR_LATENT_TASKS'
export const SET_SYNCING = 'SET_SYNCING'
export const SET_OFFLINE = 'SET_OFFLINE'
export const SET_AUTH_STATUS = 'SET_AUTH_STATUS'
export const SET_AUTH_ERROR_MESSAGE = 'SET_AUTH_ERROR_MESSAGE'
export const SET_PROPERTY = 'SET_PROPERTY'

/*
* Common
*/
export const SET_STATE = 'SET_STATE'
export const LOG_OUT = 'LOG_OUT'
export const REQUEST_AUTH = 'REQUEST_AUTH'
export const RECIEVE_AUTH = 'RECIEVE_AUTH'
export const ERROR_AUTH = 'ERROR_AUTH'

export const REQUEST_DATA = 'REQUEST_DATA'
export const RECIEVE_DATA = 'RECIEVE_DATA'
export const ERROR_DATA = 'ERROR_DATA'

export const UNDO = 'UNDO'
export const REDO = 'REDO'
export const CLEAR_UNDO_REDO = 'CLEAR_UNDO_REDO'
