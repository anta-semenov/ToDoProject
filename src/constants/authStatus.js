export const AUTH_IN_PROGRESS = 'AUTH_IN_PROGRESS'
export const AUTH_SUCESS = 'AUTH_SUCESS'
export const AUTH_ERROR = 'AUTH_ERROR'
export const AUTH_NONE = 'AUTH_NONE'
