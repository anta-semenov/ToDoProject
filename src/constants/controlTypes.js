export const TASK_TITLE_FIELD = 'TASK_TITLE_FIELD'
export const TASK_DESCRIPTION_FIELD = 'TASK_DESCRIPTION_FIELD'
