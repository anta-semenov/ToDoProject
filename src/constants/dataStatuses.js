export const DATA_NONE = 'DATA_NONE'
export const DATA_REQUESTED = 'DATA_REQUESTED'
export const DATA_ERROR = 'DATA_ERROR'
export const DATA_RECIEVED = 'DATA_RECIEVED'
