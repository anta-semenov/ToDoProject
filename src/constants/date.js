import { fromJS } from 'immutable'
export const WEEKDAY_NAMES = fromJS([
  'Sunday',
  'Monday',
  'Tuesday',
  'Wednesday',
  'Thursday',
  'Friday',
  'Saturday'
])
export const WEEKDAY_SHORT_NAMES = fromJS([
  'Su',
  'Mo',
  'Tu',
  'We',
  'Th',
  'Fr',
  'Sa'
])

export const MONTH_NAMES = fromJS([
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December'
])
export const MONTH_SHORT_NAMES = fromJS([
  'Jan',
  'Feb',
  'Mar',
  'Apr',
  'May',
  'Jun',
  'Jul',
  'Aug',
  'Sep',
  'Oct',
  'Nov',
  'Dec'
])
