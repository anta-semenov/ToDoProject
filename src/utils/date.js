export const dateDayDifference = (startDate, endDate) => {
  const dateDiff = endDate.getTime() - startDate.getTime()
  const dayDiff = dateDiff >= 0 ? Math.floor(dateDiff / (1000*60*60*24)) : Math.ceil(dateDiff / (1000*60*60*24))
  const hourLeftover = startDate < endDate ? startDate.getHours() + Math.round(dateDiff % (1000*60*60*24) / (1000*60*60)) :
  console.log(dayDiff, startDate.getHours(), Math.round(dateDiff % (1000*60*60*24) / (1000*60*60)), hourLeftover)
  return hourLeftover >= 24 ? dayDiff + 1 : dayDiff
}
export const dateHumanNameDifference = (startDate, endDate) => {

}
export const dateHumanName = (date, baseDate = Date.now()) => {

}
export const shiftDate = (date, shift, item = 'day') => {
  switch (item) {
    case 'week':
      return shiftDate(date, shift * 7)
    case 'month': {
      let yearShift = shift >= 0 ? Math.floor(shift/12) : Math.ceil(shift/12)
      let monthShift = shift % 12
      if (shift > 0 && monthShift + date.getMonth() > 11) {
        return new Date(date.getFullYear() + yearShift + 1, date.getMonth() + monthShift - 12, date.getDate(), date.getHours(), date.getMinutes(), date.getSeconds(), date.getMilliseconds())
      }
      if (shift < 0 && monthShift + date.getMonth() < 0) {
        return new Date(date.getFullYear() + yearShift - 1, date.getMonth() + monthShift + 12, date.getDate(), date.getHours(), date.getMinutes(), date.getSeconds(), date.getMilliseconds())
      }
      return new Date(date.getFullYear() + yearShift, date.getMonth() + monthShift, date.getDate(), date.getHours(), date.getMinutes(), date.getSeconds(), date.getMilliseconds())
    }
    case 'year':
      return new Date(date.getFullYear() + shift, date.getMonth(), date.getDate(), date.getHours(), date.getMinutes(), date.getSeconds(), date.getMilliseconds())
    default:
      return new Date(date.getTime() + shift * (1000*60*60*24))
  }
}
export const areDatesEqual = (date1, date2) => {
  if (date1 && date2) {
    return date1.getYear() === date2.getYear() && date1.getMonth() === date2.getMonth() && date1.getDate() === date2.getDate()
  }
  return false
}
