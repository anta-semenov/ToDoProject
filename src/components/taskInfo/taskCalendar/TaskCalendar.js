import React from 'react'
import { List } from 'immutable'
import moment from 'moment'
import { WEEKDAY_SHORT_NAMES, MONTH_NAMES } from '../../../constants/date'
import { areDatesEqual } from '../../../utils/date'
import './TaskCalendar.less'

export const weekdayNamesRow = (weekdayNames = WEEKDAY_SHORT_NAMES, weekStart = 1) => {
  return weekdayNames.reduce((weekRow, day, index) => weekRow.set(index - weekStart, day), List([0, 1, 2, 3, 4, 5, 6]))
}

export const weekdayRow = (startDate, weekStart = 1) => {
  let sd = moment(startDate)
  if (sd.day() > weekStart) {
    sd = sd.subtract(sd.day() - weekStart, 'd')
  } else if (sd.day() < weekStart) {
    sd = sd.subtract(7 - sd.day() - weekStart, 'd')
  }
  return WEEKDAY_SHORT_NAMES.map((day, index) => moment(sd).add(index, 'd').toDate())
}
export const monthWeeksList = (month, year, weekStart = 1) => {
  let startDate = new Date(year, month, 1)
  let monthArray = []
  while (startDate.getMonth() === month) {
    const week = weekdayRow(startDate, weekStart)
    monthArray.push(week)
    startDate = moment(week.get(0)).add(1, 'w').toDate()
  }
  return List(monthArray)
}

export default class TaskCalendar extends React.Component {
  constructor(props) {
    super(props)
    const today = new Date()
    this.state = {
      todayDate: new Date(today.getFullYear(), today.getMonth(), today.getDate()),
      currentMonth: this.props.selectedDate ? this.props.selectedDate.getMonth() : today.getMonth(),
      currentYear: this.props.selectedDate ? this.props.selectedDate.getFullYear() : today.getFullYear()
    }
  }
  nextPeriod() {
    if (this.state.currentMonth === 11) {
      this.setState({
        currentMonth: 0,
        currentYear: this.state.currentYear + 1
      })
    } else {
      this.setState({currentMonth: this.state.currentMonth + 1})
    }
  }
  prevPeriod() {
    if (this.state.currentMonth === 0) {
      this.setState({
        currentMonth: 11,
        currentYear: this.state.currentYear - 1
      })
    } else {
      this.setState({currentMonth: this.state.currentMonth - 1})
    }
  }
  handleDayClick(date) {
    if (areDatesEqual(date, this.props.selectedDate)) {this.props.onChange(this.props.id, undefined)}
    else {this.props.onChange(this.props.id, date.getTime())}
  }

  componentWillReceiveProps(nextProps) {
    const today = new Date()
    this.state = {
      todayDate: new Date(today.getFullYear(), today.getMonth(), today.getDate()),
      currentMonth: nextProps.selectedDate ? nextProps.selectedDate.getMonth() : today.getMonth(),
      currentYear: nextProps.selectedDate ? nextProps.selectedDate.getFullYear() : today.getFullYear()
    }
  }

  render() {
    return (
      <div className='calendar'>
        <div className='calendar__header'>
          <div className='calendar__nav'>
            <button className={`calendar__prev ${this.state.currentMonth === this.state.todayDate.getMonth() ? 'is-disabled' : ''}`} onClick={() => this.prevPeriod()}><div className='calendar__prev-sign'/></button>
            <div className='calendar__title'>
              <div className='calendar__month'>{MONTH_NAMES.get(this.state.currentMonth)}</div>
              <div className='calendar__year'>{this.state.currentYear}</div>
            </div>
            <button className='calendar__next' onClick={() => this.nextPeriod()}><div className='calendar__next-sign'/></button>
          </div>
          <div className='calendar__day-names'>
            {weekdayNamesRow().map(dayName =>
              <div key={dayName} className='calendar__day-name'>{dayName}</div>
            )}
          </div>
        </div>
        <div className='calendar__body'>
          {monthWeeksList(this.state.currentMonth, this.state.currentYear).map((week, index) =>
            <div key={index} className='calendar__week'>
              {week.map(day =>
                <div
                  key={day.getTime()}
                  className={`calendar__day
                    ${day.getMonth() < this.state.currentMonth ? 'is-prev-month' : ''}
                    ${day.getMonth() > this.state.currentMonth ? 'is-next-month' : ''}
                    ${day < this.state.todayDate ? 'is-before-today' : ''}
                    ${areDatesEqual(day, this.state.todayDate) ? 'is-today' : '' }
                    ${areDatesEqual(day, this.props.selectedDate) ? 'is-selected' : '' }
                  `}
                  onClick={() => this.handleDayClick(day)}>
                    {day.getDate()}
                </div>
              )}
            </div>
          )}
        </div>
      </div>
    )
  }
}

TaskCalendar.propTypes = {
  id: React.PropTypes.string.isRequired,
  selectedDate: React.PropTypes.instanceOf(Date),
  tasks: React.PropTypes.object,
  onChange: React.PropTypes.func.isRequired
}
