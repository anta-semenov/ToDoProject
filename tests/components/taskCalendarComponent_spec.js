import React from 'react'
import { shallow } from 'enzyme'
import { List, fromJS } from 'immutable'
import TaskCalendar, {
  weekdayRow,
  monthWeeksList,
  weekdayNamesRow
} from '../../src/components/taskInfo/taskCalendar/TaskCalendar'
import { WEEKDAY_SHORT_NAMES } from '../../src/constants/date'

describe('TaskCalendar Component', () => {
  describe('Helper functions', () => {
    describe('weekdayRow', () => {
      test('Should return correct list for startDate: Date(2016, 4, 3)', () => {
        const startDate1 = new Date(2016, 4, 3)
        const week1 = List([
          new Date(2016, 4, 2),
          new Date(2016, 4, 3),
          new Date(2016, 4, 4),
          new Date(2016, 4, 5),
          new Date(2016, 4, 6),
          new Date(2016, 4, 7),
          new Date(2016, 4, 8)
        ])
        expect(weekdayRow(startDate1)).toEqual(week1)
      })
      test('Should return correct list for startDate: Date(2016, 4, 1), weekStart: 2', () => {
        const startDate2 = new Date(2016, 4, 1)
        const week2 = List([
          new Date(2016, 3, 26),
          new Date(2016, 3, 27),
          new Date(2016, 3, 28),
          new Date(2016, 3, 29),
          new Date(2016, 3, 30),
          new Date(2016, 4, 1),
          new Date(2016, 4, 2)
        ])
        expect(weekdayRow(startDate2, 2)).toEqual(week2)
      })
    })
    describe('monthWeeksList', () => {
      test('Should return correct list for May of 2016', () => {
        const monthList = fromJS([
          [
            new Date(2016, 3, 25),
            new Date(2016, 3, 26),
            new Date(2016, 3, 27),
            new Date(2016, 3, 28),
            new Date(2016, 3, 29),
            new Date(2016, 3, 30),
            new Date(2016, 4, 1)
          ],
          [
            new Date(2016, 4, 2),
            new Date(2016, 4, 3),
            new Date(2016, 4, 4),
            new Date(2016, 4, 5),
            new Date(2016, 4, 6),
            new Date(2016, 4, 7),
            new Date(2016, 4, 8)
          ],
          [
            new Date(2016, 4, 9),
            new Date(2016, 4, 10),
            new Date(2016, 4, 11),
            new Date(2016, 4, 12),
            new Date(2016, 4, 13),
            new Date(2016, 4, 14),
            new Date(2016, 4, 15)
          ],
          [
            new Date(2016, 4, 16),
            new Date(2016, 4, 17),
            new Date(2016, 4, 18),
            new Date(2016, 4, 19),
            new Date(2016, 4, 20),
            new Date(2016, 4, 21),
            new Date(2016, 4, 22)
          ],
          [
            new Date(2016, 4, 23),
            new Date(2016, 4, 24),
            new Date(2016, 4, 25),
            new Date(2016, 4, 26),
            new Date(2016, 4, 27),
            new Date(2016, 4, 28),
            new Date(2016, 4, 29)
          ],
          [
            new Date(2016, 4, 30),
            new Date(2016, 4, 31),
            new Date(2016, 5, 1),
            new Date(2016, 5, 2),
            new Date(2016, 5, 3),
            new Date(2016, 5, 4),
            new Date(2016, 5, 5)
          ]
        ])
        expect(monthWeeksList(4, 2016)).toEqual(monthList)
      })
      test('Should return correct list for May of 2016 and startWeek 2', () => {
        const monthList = fromJS([
          [
            new Date(2016, 3, 26),
            new Date(2016, 3, 27),
            new Date(2016, 3, 28),
            new Date(2016, 3, 29),
            new Date(2016, 3, 30),
            new Date(2016, 4, 1),
            new Date(2016, 4, 2)
          ],
          [
            new Date(2016, 4, 3),
            new Date(2016, 4, 4),
            new Date(2016, 4, 5),
            new Date(2016, 4, 6),
            new Date(2016, 4, 7),
            new Date(2016, 4, 8),
            new Date(2016, 4, 9)
          ],
          [
            new Date(2016, 4, 10),
            new Date(2016, 4, 11),
            new Date(2016, 4, 12),
            new Date(2016, 4, 13),
            new Date(2016, 4, 14),
            new Date(2016, 4, 15),
            new Date(2016, 4, 16)
          ],
          [
            new Date(2016, 4, 17),
            new Date(2016, 4, 18),
            new Date(2016, 4, 19),
            new Date(2016, 4, 20),
            new Date(2016, 4, 21),
            new Date(2016, 4, 22),
            new Date(2016, 4, 23)
          ],
          [
            new Date(2016, 4, 24),
            new Date(2016, 4, 25),
            new Date(2016, 4, 26),
            new Date(2016, 4, 27),
            new Date(2016, 4, 28),
            new Date(2016, 4, 29),
            new Date(2016, 4, 30)
          ],
          [
            new Date(2016, 4, 31),
            new Date(2016, 5, 1),
            new Date(2016, 5, 2),
            new Date(2016, 5, 3),
            new Date(2016, 5, 4),
            new Date(2016, 5, 5),
            new Date(2016, 5, 6)
          ]
        ])
        expect(monthWeeksList(4, 2016, 2)).toEqual(monthList)
      })
      test('Should return correct list for November of 2016 and startWeek 1', () => {
        const monthList = fromJS([
          [
            new Date(2016, 9, 31),
            new Date(2016, 10, 1),
            new Date(2016, 10, 2),
            new Date(2016, 10, 3),
            new Date(2016, 10, 4),
            new Date(2016, 10, 5),
            new Date(2016, 10, 6)
          ],
          [
            new Date(2016, 10, 7),
            new Date(2016, 10, 8),
            new Date(2016, 10, 9),
            new Date(2016, 10, 10),
            new Date(2016, 10, 11),
            new Date(2016, 10, 12),
            new Date(2016, 10, 13)
          ],
          [
            new Date(2016, 10, 14),
            new Date(2016, 10, 15),
            new Date(2016, 10, 16),
            new Date(2016, 10, 17),
            new Date(2016, 10, 18),
            new Date(2016, 10, 19),
            new Date(2016, 10, 20)
          ],
          [
            new Date(2016, 10, 21),
            new Date(2016, 10, 22),
            new Date(2016, 10, 23),
            new Date(2016, 10, 24),
            new Date(2016, 10, 25),
            new Date(2016, 10, 26),
            new Date(2016, 10, 27)
          ],
          [
            new Date(2016, 10, 28),
            new Date(2016, 10, 29),
            new Date(2016, 10, 30),
            new Date(2016, 11, 1),
            new Date(2016, 11, 2),
            new Date(2016, 11, 3),
            new Date(2016, 11, 4)
          ]
        ])
        expect(monthWeeksList(10, 2016, 1)).toEqual(monthList)
      })
    })
    describe('weekdayNamesRow', () => {
      test('Should return correct list of weekday names for weekstart at Monday (1)', () => {
        const weekdayNames = List(['Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa', 'Su'])
        expect(weekdayNamesRow()).toEqual(weekdayNames)
      })
      test('Should return correct list of weekday names for weekstart at Wednesday (3)', () => {
        const weekdayNames = List(['We', 'Th', 'Fr', 'Sa', 'Su', 'Mo', 'Tu'])
        expect(weekdayNamesRow(WEEKDAY_SHORT_NAMES, 3)).toEqual(weekdayNames)
      })
    })
  })
  describe('Component', () => {
    test('Should render all elements of component: navigation with prev and next button, title and weekday names. Also it should render dates', () => {
      const calendarComponent = shallow(<TaskCalendar id="djsdag23" onChange={() => {}} />)
      expect(calendarComponent).toMatchSnapshot()
      // const calendarNavigation = findRenderedDOMComponentWithClass(calendarComponent, 'calendar__nav')
      // const calendarMonth = findRenderedDOMComponentWithClass(calendarComponent, 'calendar__month')
      // const calendarYear = findRenderedDOMComponentWithClass(calendarComponent, 'calendar__year')
      // const prevBtn = findRenderedDOMComponentWithClass(calendarComponent, 'calendar__prev')
      // const nextBtn = findRenderedDOMComponentWithClass(calendarComponent, 'calendar__next')
      // const weekdayNames = findRenderedDOMComponentWithClass(calendarComponent, 'calendar__day-names')
      // const weekdayNameArray = scryRenderedDOMComponentsWithClass(calendarComponent, 'calendar__day-name')
      // const calendarDayArray = scryRenderedDOMComponentsWithClass(calendarComponent, 'calendar__day')
      //
      // expect(calendarNavigation.className).toEqual('calendar__nav')
      // expect(calendarMonth.className).toEqual('calendar__month')
      // expect(calendarYear.className).toEqual('calendar__year')
      // expect(prevBtn.className).toContain('calendar__prev')
      // expect(nextBtn.className).toEqual('calendar__next')
      // expect(weekdayNames.className).toEqual('calendar__day-names')
      // expect(weekdayNameArray.length).toEqual(7)
      // expect(calendarDayArray.length).toBeGreaterThan(27)
      // expect(calendarDayArray.length).toBeLessThan(43)
    })
  })
})
