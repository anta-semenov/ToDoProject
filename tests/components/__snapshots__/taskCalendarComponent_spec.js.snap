// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskCalendar Component Component Should render all elements of component: navigation with prev and next button, title and weekday names. Also it should render dates 1`] = `
<div
  className="calendar"
>
  <div
    className="calendar__header"
  >
    <div
      className="calendar__nav"
    >
      <button
        className="calendar__prev is-disabled"
        onClick={[Function]}
      >
        <div
          className="calendar__prev-sign"
        />
      </button>
      <div
        className="calendar__title"
      >
        <div
          className="calendar__month"
        >
          March
        </div>
        <div
          className="calendar__year"
        >
          2018
        </div>
      </div>
      <button
        className="calendar__next"
        onClick={[Function]}
      >
        <div
          className="calendar__next-sign"
        />
      </button>
    </div>
    <div
      className="calendar__day-names"
    >
      <div
        className="calendar__day-name"
        key="Mo"
      >
        Mo
      </div>
      <div
        className="calendar__day-name"
        key="Tu"
      >
        Tu
      </div>
      <div
        className="calendar__day-name"
        key="We"
      >
        We
      </div>
      <div
        className="calendar__day-name"
        key="Th"
      >
        Th
      </div>
      <div
        className="calendar__day-name"
        key="Fr"
      >
        Fr
      </div>
      <div
        className="calendar__day-name"
        key="Sa"
      >
        Sa
      </div>
      <div
        className="calendar__day-name"
        key="Su"
      >
        Su
      </div>
    </div>
  </div>
  <div
    className="calendar__body"
  >
    <div
      className="calendar__week"
      key="0"
    >
      <div
        className="calendar__day
                    is-prev-month
                    
                    is-before-today
                    
                    
                  "
        key="1519621200000"
        onClick={[Function]}
      >
        26
      </div>
      <div
        className="calendar__day
                    is-prev-month
                    
                    is-before-today
                    
                    
                  "
        key="1519707600000"
        onClick={[Function]}
      >
        27
      </div>
      <div
        className="calendar__day
                    is-prev-month
                    
                    is-before-today
                    
                    
                  "
        key="1519794000000"
        onClick={[Function]}
      >
        28
      </div>
      <div
        className="calendar__day
                    
                    
                    is-before-today
                    
                    
                  "
        key="1519880400000"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="calendar__day
                    
                    
                    is-before-today
                    
                    
                  "
        key="1519966800000"
        onClick={[Function]}
      >
        2
      </div>
      <div
        className="calendar__day
                    
                    
                    is-before-today
                    
                    
                  "
        key="1520053200000"
        onClick={[Function]}
      >
        3
      </div>
      <div
        className="calendar__day
                    
                    
                    is-before-today
                    
                    
                  "
        key="1520139600000"
        onClick={[Function]}
      >
        4
      </div>
    </div>
    <div
      className="calendar__week"
      key="1"
    >
      <div
        className="calendar__day
                    
                    
                    is-before-today
                    
                    
                  "
        key="1520226000000"
        onClick={[Function]}
      >
        5
      </div>
      <div
        className="calendar__day
                    
                    
                    is-before-today
                    
                    
                  "
        key="1520312400000"
        onClick={[Function]}
      >
        6
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    is-today
                    
                  "
        key="1520398800000"
        onClick={[Function]}
      >
        7
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1520485200000"
        onClick={[Function]}
      >
        8
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1520571600000"
        onClick={[Function]}
      >
        9
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1520658000000"
        onClick={[Function]}
      >
        10
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1520744400000"
        onClick={[Function]}
      >
        11
      </div>
    </div>
    <div
      className="calendar__week"
      key="2"
    >
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1520827200000"
        onClick={[Function]}
      >
        12
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1520913600000"
        onClick={[Function]}
      >
        13
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1521000000000"
        onClick={[Function]}
      >
        14
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1521086400000"
        onClick={[Function]}
      >
        15
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1521172800000"
        onClick={[Function]}
      >
        16
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1521259200000"
        onClick={[Function]}
      >
        17
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1521345600000"
        onClick={[Function]}
      >
        18
      </div>
    </div>
    <div
      className="calendar__week"
      key="3"
    >
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1521432000000"
        onClick={[Function]}
      >
        19
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1521518400000"
        onClick={[Function]}
      >
        20
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1521604800000"
        onClick={[Function]}
      >
        21
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1521691200000"
        onClick={[Function]}
      >
        22
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1521777600000"
        onClick={[Function]}
      >
        23
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1521864000000"
        onClick={[Function]}
      >
        24
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1521950400000"
        onClick={[Function]}
      >
        25
      </div>
    </div>
    <div
      className="calendar__week"
      key="4"
    >
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1522036800000"
        onClick={[Function]}
      >
        26
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1522123200000"
        onClick={[Function]}
      >
        27
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1522209600000"
        onClick={[Function]}
      >
        28
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1522296000000"
        onClick={[Function]}
      >
        29
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1522382400000"
        onClick={[Function]}
      >
        30
      </div>
      <div
        className="calendar__day
                    
                    
                    
                    
                    
                  "
        key="1522468800000"
        onClick={[Function]}
      >
        31
      </div>
      <div
        className="calendar__day
                    
                    is-next-month
                    
                    
                    
                  "
        key="1522555200000"
        onClick={[Function]}
      >
        1
      </div>
    </div>
  </div>
</div>
`;
